controllers/controllers/workloads/integration/cfspace_controller_integration_test.go

```diff
@@ -295,25 +296,127 @@ var _ = Describe("CFSpaceReconciler Integration Tests", func() {
 				return k8sClient.Get(ctx, types.NamespacedName{Name: rootServiceAccount.Name, Namespace: cfSpace.Name}, &propagatedServiceAccount)
 			}).Should(Succeed())
 
-			// Simulate k8s adding a token secret to the propagated service account
+			// Simulate k8s adding a token secret to the propagated service account AND the propagated service account having a stale image registry credential secret
 			origPropagatedServiceAccount := propagatedServiceAccount.DeepCopy()
-			propagatedServiceAccount.Secrets = append(propagatedServiceAccount.Secrets, corev1.ObjectReference{Name: "some-token-secret"})
+			tokenSecretName = rootServiceAccount.Name + "-token-XYZABC"
+			propagatedServiceAccount.Secrets = []corev1.ObjectReference{{Name: tokenSecretName}, {Name: "out-of-date-registry-credentials"}}
 			Expect(k8sClient.Patch(ctx, &propagatedServiceAccount, client.MergeFrom(origPropagatedServiceAccount))).To(Succeed())
 
-			// Add a new secret to the root copy of the service account
+			// Modify the root service account to trigger reconciliation
 			origRootServiceAccount := rootServiceAccount.DeepCopy()
-			rootServiceAccount.Secrets = append(rootServiceAccount.Secrets, corev1.ObjectReference{Name: "ignore-me-please"})
+			rootServiceAccount.Labels = map[string]string{"new-label": "dummy-value"}
 			Expect(k8sClient.Patch(ctx, rootServiceAccount, client.MergeFrom(origRootServiceAccount))).To(Succeed())
 		})
 
 		It("doesn't modify the secrets on the propagated service account", func() {
-			Consistently(func(g Gomega) {
+			Eventually(func(g Gomega) {
+				var updatedPropagatedServiceAccount corev1.ServiceAccount
+				g.Expect(
+					k8sClient.Get(ctx, client.ObjectKeyFromObject(&propagatedServiceAccount), &updatedPropagatedServiceAccount),
+				).To(Succeed())
+				g.Expect(updatedPropagatedServiceAccount.Secrets).To(ConsistOf(
+					corev1.ObjectReference{Name: tokenSecretName},
+					corev1.ObjectReference{Name: packageRegistrySecretName},
+				))
+				g.Expect(updatedPropagatedServiceAccount.ImagePullSecrets).To(Equal(rootServiceAccount.ImagePullSecrets))
+			}).Should(Succeed())
+		})
+	})
+
+	When("the package registry secret is added to the root service account", func() {
+		var (
+			rootServiceAccount       *corev1.ServiceAccount
+			propagatedServiceAccount corev1.ServiceAccount
+			tokenSecretName          string
+		)
+		BeforeEach(func() {
```

Comment:
There is quite a substantial repetition in the `BeforeEach` setup of the `the package registry secret is added to the root service account` and `the image pull secrets are updated on the root service account`. Would it make sense to group those two under a `Describe("Updating the service account secrets"...)` so that the two contexts just configure the service account in their `BeforeEach`?

Reply:
We thought about that, but each of the tests differs in subtle ways from the others, so it felt like it wasn't worth the complexity to try to refactor it while accommodating all of the slightly different behaviors we still need.